docs/source/module_configuration/staticsite.rst

```diff
@@ -111,3 +111,149 @@ Most of these options are not required, but are listed here for reference::
+
+.. image:: ../images/staticsite/auth_at_edge/flow_diagram.png
+
+Here is how the solution works:
+
+    1. The viewer’s web browser is redirected to Amazon Cognito custom UI page to sign up and authenticate.
+    2. After authentication, Cognito generates and cryptographically signs a JWT then responds with a redirect containing the JWT embedded in the URL.
+    3. The viewer’s web browser extracts JWT from the URL and makes a request to private content (private/* path), adding Authorization request header with JWT.
+    4. Amazon CloudFront routes the request to the nearest AWS edge location. The CloudFront distribution’s private behavior is configured to launch a `Lambda@Edge` function on ViewerRequest event.
+    5. `Lambda@Edge` decodes the JWT and checks if the user belongs to the correct Cognito User Pool. It also verifies the cryptographic signature using the public RSA key for Cognito User Pool. Crypto verification ensures that JWT was created by the trusted party.
+    6. After passing all of the verification steps, `Lambda@Edge` strips out the Authorization header and allows the request to pass through to designated origin for CloudFront. In this case, the origin is the private content Amazon S3 bucket.
+    7. After receiving response from the origin S3 bucket, CloudFront sends the response back to the browser. The browser displays the data from the returned response.
+
+An example of a Auth@Edge static site configuration is as follows. All listed options are required::
+
+    deployments:
+      - modules:
+        - name: sample-app
+          path: sample-app
+          type: static
+          environments:
+            dev:
+              namespace: sample-app-dev
+              staticsite_auth_at_edge: true
+              staticsite_aae_user_pool_arn: arn:aws:cognito-idp:us-east-1:240134083525:userpool/us-east-1_cjVgcUyWV
+        regions:
+          # NOTE: At the time of this writing Lambda@EDGE is only
+          # available in us-east-1. This solution will fail in
+          # other regions
+          - us-east-1
+
+The `Auth@Edge` functionality does not create a Cognito User Pool for you.
+This is a manual step that needs to be completed prior to launching
+the stack as a user pool arn is required.  A client will be automatically
+created within this user pool for use with the application.
+
+Example of `Auth@Edge` Options
+----------------------------------
+
+The following example lists all the configurable elements of an `Auth@Edge` stack.
+These work in conjunction with those listed above for a standard static site in
+most circumstances. All examples of Optional components represent the default
+assignment::
+
+    deployments:
+      - modules:
+          - name: auth-at-edge
+            path: web
+            class_path: runway.module.staticsite.StaticSite
+            parameters:
+              namespace: aae-${env DEPLOY_ENVIRONMENT}
+
+              # (Required), indicate that this stack should utilize
```

Comment:
can we drop the _documentation comments in code yaml code blocks_ and create a **Parameters** section in the docs? I don't think we need to separate auth@edge parameters from the rest but document them all in one top-level **Parameters** section. I would even make it the first section of the page.

Format example from cfngin hooks - https://docs.onica.com/projects/runway/en/release/cfngin/hooks.html#aws-lambda-upload-lambda-functions

```
Static Site
===========


Parameters
------------

**namespace (str)**
    description here...
```

Reply:
I'm on the fence with this one, especially now that we changed the variables from `staticsite_aae` to `staticsite`. I'm worried that the onslaught of all these variables that don't pertain to the main `staticsite` module may confuse the reader. The two sections are a little cumbersome but logically separate the two items into distinct module types.

@troyready, thoughts?

Reply:
Talked with @troyready, agrees with combining the parameters. I'll work on updating.